staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -141,7 +159,12 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 			continue // todo: wire oldObj parameter
 		}
 		evalResult, evalDetails, err := compiled.Program.Eval(activation)
-
+		rtCost := evalDetails.ActualCost()
+		if *rtCost > math.MaxInt || int64(*rtCost) > remainingBudget {
+			errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("validation failed due to cost budget exceed")))
```

Comment:
Report what the budget is how much it was exceeded by? (We might need to keep track of more information to give developer friendly errors here?)

Reply:
I will add into the error message